src/collection/components/CollectionFragment.tsx

```diff
@@ -65,35 +68,64 @@ const CollectionFragment: FunctionComponent<CollectionFragmentProps> = ({
 		/>
 	);
 
-	const renderForm = (fragment: any, index: number) => (
-		<Form>
-			<FormGroup label={`Tekstblok titel`} labelFor="title">
-				<TextInput
-					id="title"
-					type="text"
-					value={fragment.custom_title || ''}
-					placeholder="Titel"
-					onChange={(value: string) => updateFragmentProperty(value, 'custom_title', fragment.id)}
-				/>
-			</FormGroup>
-			<FormGroup label={`Tekstblok beschrijving`} labelFor={`beschrijving_${index}`}>
-				<>
+	const onChangeToggle = () => {
+		updateFragmentProperty(!useCustomFields, 'use_custom_fields', fragment.id);
+		setUseCustomFields(!useCustomFields);
+	};
+
+	const renderForm = (fragment: any, itemMeta: any, index: number) => {
+		const onChangeTitle = (value: string) => {
+			if (itemMeta && !useCustomFields) {
+				return null;
+			}
+
+			return updateFragmentProperty(value, 'custom_title', fragment.id);
+		};
+
+		const onChangeDescription = (e: any) => {
+			if (itemMeta && !useCustomFields) {
+				return null;
+			}
+
+			return updateFragmentProperty(e.target.innerHTML, 'custom_description', fragment.id);
+		};
+
+		return (
+			<Form>
+				{itemMeta && <Toggle checked={useCustomFields} onChange={onChangeToggle} />}
+				<FormGroup label={`Tekstblok titel`} labelFor="title">
+					<TextInput
+						id="title"
+						type="text"
+						value={
```

Comment:
Hier ben ik niet super blij mee. Is er een logischere manier om dit te schrijven waar ik totaal over kijk?

Reply:
maybe:
```
getFragmentProperty(itemMeta, fragment, 'title');
getFragmentProperty(itemMeta, fragment, 'description');
```

```
const getFragmentProperty(itemMeta, fragment, prop) {
  if (useCustomFields) {
      return get(fragment, 'custom_' + prop) || get(itemMeta, prop) || '';
  } else {
      return get(itemMeta, prop) || '';
  }
}
```

7 regels ipv 10 :p